File: go/lasagna-master/lasagna_master.go
package lasagna

func PreparationTime(layers []string, timePerLayer int) int {
    if timePerLayer == 0 {
        return len(layers) * 2
    }
	return len(layers) * timePerLayer
}

func Quantities(layers []string) (int, float64) {
    noodles := 0
    sauce := 0.0
    for _, ingredient := range layers {
        if ingredient == "noodles" {
            noodles += 50
        } else if ingredient == "sauce" {
        	sauce += 0.2
        }
    }
	return noodles, sauce
}

func AddSecretIngredient(friendsList []string, myList []string) []string {
    myList[len(myList)-1] = friendsList[len(friendsList) - 1]
    return myList
}

func ScaleRecipe(quantities []float64, numberOfPortions int) []float64 {
    var newQuantities = make([]float64, len(quantities))
    scalePortions := float64(numberOfPortions)/2
    for i := 0; i < len(quantities); i++ {
        newQuantities[i] = quantities[i] * scalePortions
    }
	return newQuantities
}

File: javascript/lasagna-master/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */

export function cookingStatus(remainingTime) {
  switch (remainingTime) {
    case undefined:
      return 'You forgot to set the timer.';
    case 0:
      return 'Lasagna is done.';
    default:
      return 'Not done, please wait.';
  };
}

export function preparationTime(layers, timePerLayer = 2) {
  return layers.length * timePerLayer;
}

export function quantities(layers) {
  let amounts = {
    noodles: 0,
    sauce: 0,
  };
  for (let i = 0; i < layers.length; i++) {
    if (layers[i] === 'sauce') {
      amounts.sauce += 0.2;
    } else if (layers[i] === 'noodles') {
      amounts.noodles += 50;
    };
  };
  return amounts;
}

export function addSecretIngredient(ingredients1, ingredients2) {
  ingredients2.push(ingredients1[ingredients1.length - 1]);
}

export function scaleRecipe(recipe, portions) {
  let newRecipe = {};
  for (let key in recipe) {
    newRecipe[key] = recipe[key] / 2 * portions;
  };
  return newRecipe;
}
